src/AsyncEventSource.cpp

```diff
@@ -247,14 +248,33 @@ void AsyncEventSourceClient::_runQueue(){
   this->_messageQueue_processing = true;
 #endif // ESP32
 
-  while(!_messageQueue.isEmpty() && _messageQueue.front()->finished()){
-    _messageQueue.remove(_messageQueue.front());
-  }
-
+  size_t total_bytes_written = 0;
   for(auto i = _messageQueue.begin(); i != _messageQueue.end(); ++i)
   {
-    if(!(*i)->sent())
-      (*i)->send(_client);
+    if(!(*i)->sent()) {
+      size_t bytes_written = (*i)->write_buffer(_client);
+      total_bytes_written += bytes_written;
+      if(bytes_written == 0)
+        break;
+      // todo: there is a further optimization to write a partial event to squeeze the last few bytes into the outgoing tcp send buffer, in 
+      //           fact all of this code is already set up to do so, it's only write_buffer that needs to be updated to allow it instead of 
+      //           returning zero when the full event won't fit into what's left of the buffer
+      // todo: windows is taking 40-50ms to send an ack back while it waits for more data which won't come since this code must wait for ack first 
```

Comment:
realized after the fact this comment is pointless since the eventing protocol is unidirectional